File: src/main/java/com/shockang/study/algorithm/java/leetcode/leetcode201_300/leetcode215/solution1/Solution.java
package com.shockang.study.algorithm.java.leetcode.leetcode201_300.leetcode215.solution1;

import java.util.PriorityQueue;

/**
 * 优先队列
 *
 * @author Shockang
 */
public class Solution {
	public int findKthLargest(int[] nums, int k) {
		PriorityQueue<Integer> priorityQueue = new PriorityQueue<>();
		for (int num : nums) {
			priorityQueue.add(num);
		}
		int n = nums.length;
		for (int i = 0; i < n - k; i++) {
			priorityQueue.poll();
		}
		return priorityQueue.peek();
	}
}

File: src/main/scala/com/shockang/study/algorithm/scala/leetcode/leetcode201_300/leetcode215/solution1/Solution.scala
package com.shockang.study.algorithm.scala.leetcode.leetcode201_300.leetcode215.solution1

import scala.collection.mutable

/**
 *
 * @author Shockang
 */
object Solution {
  def findKthLargest(nums: Array[Int], k: Int): Int = {
    val priorityQueue = mutable.PriorityQueue[Int]()
    nums.foreach(priorityQueue.enqueue(_))
    for (_ <- 0 until k - 1) priorityQueue.dequeue()
    priorityQueue.dequeue()
  }
}